### PR TITLE
[IMP] l10n_ar_account_withholding_cba: new module

### PR DESCRIPTION
**Pull request by pablohmontenegro**

### CODE REVIEW

**Comment by roboadhoc** [comment]
[Pull request status dashboard](https://mergebot.adhoc.ar/ingadhoc/odoo-argentina/pull/900).

File: l10n_ar_account_withholding_cba/models/account_tax.py [hunk 1]
```diff
@@ -0,0 +1,33 @@
+from odoo import fields, models, api
+
+
+class AccountTax(models.Model):
+    _inherit = 'account.tax'
+
+    ratio = fields.Float(required=True, default=100, help="Ratio to apply to tax base amount")
```
Comment (pablohmontenegro) [inline l10n_ar_account_withholding_cba/models/account_tax.py]: Para quien haga review técnico me queda dudas si debe ser requerido, o si debemos hacer que siempre sea 100 y hacemos que sea solo visible para argentina

File: l10n_ar_account_withholding_cba/README.rst [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+
+.. |icon| image:: https://raw.githubusercontent.com/ingadhoc/maintainer-tools/master/resources/adhoc-icon.png
+
+.. image:: https://img.shields.io/badge/license-AGPL--3-blue.png
+   :target: https://www.gnu.org/licenses/agpl
+   :alt: License: AGPL-3
+
+=============================
+Tax Settlements For Cordoba
+=============================
```
Comment (zaoral) [inline l10n_ar_account_withholding_cba/README.rst]: the length of the === should be the same of the text, if not will get a warning un runbot for error in the manifest rst file

File: l10n_ar_account_withholding_cba/models/account_tax.py [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+        return vals
+
+    def _compute_amount(
+            self, base_amount, price_unit, quantity, product, partner=None, fixed_multiplicator=1):
+        res = super()._compute_amount(base_amount, price_unit, quantity, product, partner=partner, fixed_multiplicator=fixed_multiplicator)
+        if self.amount_type == 'partner_tax' and self.ratio:
+            date = self._context.get('invoice_date') or fields.Date.context_today(self)
+            partner = partner and partner.sudo()
+            return base_amount * self.sudo().get_partner_alicuota_percepcion(partner, date) * self.ratio / 100
+        return res
```
Comment (zaoral) [inline l10n_ar_account_withholding_cba/models/account_tax.py]: return  super()._compute_amount(base_amount, price_unit, quantity, product, partner=partner, fixed_multiplicator=fixed_multiplicator)

File: l10n_ar_account_withholding_cba/models/account_tax.py [hunk 4]
```diff
@@ -0,0 +1,30 @@
...
+    def get_withholding_vals(self, payment_group):
+        vals = super().get_withholding_vals(payment_group)
+        if self.withholding_type == 'partner_tax' and self.ratio:
+            vals['withholdable_base_amount'] *= self.ratio / 100
+            vals['period_withholding_amount'] *= self.ratio / 100
+        return vals
+
+    def _compute_amount(
+            self, base_amount, price_unit, quantity, product, partner=None, fixed_multiplicator=1):
+        res = super()._compute_amount(base_amount, price_unit, quantity, product, partner=partner, fixed_multiplicator=fixed_multiplicator)
```
Comment (zaoral) [inline l10n_ar_account_withholding_cba/models/account_tax.py]: you do not change result of the super, so i think you can remove it here and simple call it at the return

**Review by zaoral — CHANGES_REQUESTED** [review]

File: l10n_ar_account_withholding_ratio/views/account_tax_view.xml [hunk 5]
```diff
@@ -0,0 +1,16 @@
+<?xml version="1.0" encoding="utf-8"?>
+<odoo>
+
+    <record id="view_tax_form" model="ir.ui.view">
+        <field name="name">account.tax.form</field>
+        <field name="model">account.tax</field>
+        <field name="inherit_id" ref="account_withholding_automatic.view_tax_form"/>
+        <field name="arch" type="xml">
+            <field name="withholding_user_error_message" position="after">
+                <field name="ratio" attrs="{'invisible':[('country_code', '!=', 'AR')]}"/>
```
Comment (zaoral) [inline l10n_ar_account_withholding_ratio/views/account_tax_view.xml]:
```suggestion
                <div attrs="{'invisible':[('country_code', '!=', 'AR')]}">
                    <label name="ratio">
                    <field name="ratio"/>
                    %
                </div>
```

**Review by zaoral — APPROVED** [review]
rename to l10n_ar_account_withh

**Comment by zaoral** [comment]
@roboadhoc r+

**Comment by fw-bot-adhoc** [comment]
@pablohmontenegro @zaoral this pull request has forward-port PRs awaiting action (not merged or closed):
- ingadhoc/odoo-argentina#948

**Comment by fw-bot-adhoc** [comment]
@pablohmontenegro @zaoral this pull request has forward-port PRs awaiting action (not merged or closed):
- ingadhoc/odoo-argentina#948

**Comment by fw-bot-adhoc** [comment]
@pablohmontenegro @zaoral this pull request has forward-port PRs awaiting action (not merged or closed):
- ingadhoc/odoo-argentina#948

**Comment by fw-bot-adhoc** [comment]
@pablohmontenegro @zaoral this pull request has forward-port PRs awaiting action (not merged or closed):
- ingadhoc/odoo-argentina#948

**Comment by fw-bot-adhoc** [comment]
@pablohmontenegro @zaoral this pull request has forward-port PRs awaiting action (not merged or closed):
- ingadhoc/odoo-argentina#948

**Comment by fw-bot-adhoc** [comment]
@pablohmontenegro @zaoral this pull request has forward-port PRs awaiting action (not merged or closed):
- ingadhoc/odoo-argentina#948

**Comment by fw-bot-adhoc** [comment]
@pablohmontenegro @zaoral this pull request has forward-port PRs awaiting action (not merged or closed):
- ingadhoc/odoo-argentina#948

**Comment by fw-bot-adhoc** [comment]
@pablohmontenegro @zaoral this pull request has forward-port PRs awaiting action (not merged or closed):
- ingadhoc/odoo-argentina#948

**Comment by fw-bot-adhoc** [comment]
@pablohmontenegro @zaoral this pull request has forward-port PRs awaiting action (not merged or closed):
- ingadhoc/odoo-argentina#948

**Comment by fw-bot-adhoc** [comment]
@pablohmontenegro @zaoral this pull request has forward-port PRs awaiting action (not merged or closed):
- ingadhoc/odoo-argentina#948

**Comment by fw-bot-adhoc** [comment]
@pablohmontenegro @zaoral this pull request has forward-port PRs awaiting action (not merged or closed):
- ingadhoc/odoo-argentina#948

**Pull request closed without merging**